io-sim/src/Control/Monad/IOSim/Types.hs

```diff
@@ -870,9 +896,22 @@ data StmTxResult s a =
      | StmTxAborted  [SomeTVar s] SomeException
 
 
--- | OrElse/Catch give rise to an alternate right hand side branch. A right branch
--- can be a NoOp
-data BranchStmA s a = OrElseStmA (StmA s a) | NoOpStmA
+-- | A branch indicates an alternative statment available in the current context.
+-- For example, `OrElse` has two alternative statements, say "left" and "right".
+-- While executing the left statement, `OrElseStmA` branch indicates that the
+-- right branch is still available, in case, left statement fails.
+data BranchStmA s a =
+       -- | `OrElse` statement with its 'right' alternative.
+       OrElseStmA (StmA s a)
+       -- | `CatchStm` statment with 'catch' handler.
+     | CatchStmA (SomeException -> StmA s a)
+       -- | No op statment is not a part of the STM semantics. It simply
```

Comment:
`No op statment is not a part of the STM semantics.` -> `Unlike the other two branches, the no-op branch is not an explicit part of the STM syntax.`